services/venice-controller/src/main/java/com/linkedin/venice/controller/kafka/consumer/ControllerKafkaClientFactory.java

```diff
@@ -41,6 +45,11 @@ public Properties setupSSL(Properties properties) {
     } else {
       properties.setProperty(CommonClientConfigs.BOOTSTRAP_SERVERS_CONFIG, controllerConfig.getKafkaBootstrapServers());
     }
+    if (!StringUtils.isBlank(controllerConfig.getKafkaSaslMechanism())) {
+      properties.setProperty(CommonClientConfigs.SECURITY_PROTOCOL_CONFIG, controllerConfig.getKafkaSecurityProtocol());
+      properties.setProperty("sasl.mechanism", controllerConfig.getKafkaSaslMechanism());
```

Comment:
is it possible that we put these configs inside the `SSLConfig`?  Then we can directly copy all stuff there with:
`properties.putAll(sslConfig.get().getKafkaSSLConfig());`